diff --git a/src/extensions/renderer/canvas/index.js b/src/extensions/renderer/canvas/index.js
index a959ecc2..edc4cca0 100644
--- a/src/extensions/renderer/canvas/index.js
+++ b/src/extensions/renderer/canvas/index.js
@@ -50,7 +50,6 @@ function CanvasRenderer( options ){
     bufferContexts: new Array( CRp.CANVAS_LAYERS ),
   };
 
-  var tapHlOff = '-webkit-tap-highlight-color: rgba(0,0,0,0);';
   var tapHlOffAttr = '-webkit-tap-highlight-color';
   var tapHlOffStyle = 'rgba(0,0,0,0)';
   r.data.canvasContainer = document.createElement( 'div' ); // eslint-disable-line no-undef
@@ -62,10 +61,7 @@ function CanvasRenderer( options ){
 
   var container = options.cy.container();
   container.appendChild( r.data.canvasContainer );
-
-  if( (container.getAttribute('style') || '').indexOf(tapHlOff) < 0 ){
-    container.style[tapHlOffAttr] = tapHlOffStyle;
-  }
+  container.style[tapHlOffAttr] = tapHlOffStyle;
 
   var styleMap = {
     '-webkit-user-select': 'none',
@@ -73,19 +69,19 @@ function CanvasRenderer( options ){
     'user-select': 'none',
     '-webkit-tap-highlight-color': 'rgba(0,0,0,0)',
     'outline-style': 'none',
-  }
+  };
 
   if(is.ms()) {
-    styleMap['-ms-touch-action'] = 'none'
-    styleMap['touch-action'] = 'none'
+    styleMap['-ms-touch-action'] = 'none';
+    styleMap['touch-action'] = 'none';
   }
 
   for( var i = 0; i < CRp.CANVAS_LAYERS; i++ ){
     var canvas = r.data.canvases[ i ] = document.createElement( 'canvas' );  // eslint-disable-line no-undef
     r.data.contexts[ i ] = canvas.getContext( '2d' );
-    for(const [k,v] of Object.entries(styleMap)) {
-      canvas.style[k] = v;
-    }
+    Object.keys(styleMap).forEach((k) => {
+      canvas.style[k] = styleMap[k];
+    });
     canvas.style.position = 'absolute';
     canvas.setAttribute( 'data-id', 'layer' + i );
     canvas.style.zIndex = String( CRp.CANVAS_LAYERS - i );
